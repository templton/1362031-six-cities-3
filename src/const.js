export const PlaceCardTypes = {
  APARTMENT: `apartment`,
  ROOM: `room`,
  HOUSE: `house`,
  HOTEL: `hotel`,
};

export const sortCityFilterType = {
  POPULAR: `Popular`,
  TOP_RATED_FIRST: `Top rated first`,
  PRICE_LOW_TO_HIGH: `Price: low to high`,
  PRICE_HIGH_TO_LOW: `Price: high to low`,
};

export const routerPath = {
  MAIN: `/`,
  LOGIN: `/login`,
  FAVOURITES: `/favorites`,
  HOTEL_DETAIL: `/hotel/:hotelId`
};
