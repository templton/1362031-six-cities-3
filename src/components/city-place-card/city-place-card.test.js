import React from "react";
import renderer from "react-test-renderer";
import CityPlaceCard from "./city-place-card";
import {CardClasses} from "../../const";

const info = {
  id: 1,
  title: `Комната №1`,
  isPremium: false,
  image: `https://upload.wikimedia.org/wikipedia/commons/3/34/Nanteos_%28geograph_3153536%29.jpg`,
  costPerNignt: 80,
  type: `room`,
  raiting: 4.7
};

describe(`CityPlaceCard render`, () => {

  it(`Render CityPlaceCard`, () => {
    const onClickCardTitle = jest.fn();
    const tree = renderer
      .create(<CityPlaceCard
        info={info}
        cardClass={CardClasses.CITIES}
        onCityPlaceCardMouseEnter={()=>{}}
        onClickCardTitle={onClickCardTitle}/>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
