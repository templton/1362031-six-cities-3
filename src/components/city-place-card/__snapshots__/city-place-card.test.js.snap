// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityPlaceCard render Render CityPlaceCard 1`] = `
<article
  className="cities__placecard place-card"
  onMouseEnter={[Function]}
>
  <div
    className="place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="https://upload.wikimedia.org/wikipedia/commons/3/34/Nanteos_%28geograph_3153536%29.jpg"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          80
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button place-card__bookmark-button button"
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          In bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "80%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
      onClick={[Function]}
    >
      <a
        href="#"
      >
        Комната №1
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      room
    </p>
  </div>
</article>
`;
